core/src/main/groovy/noe/tomcat/configure/ConnectorAttributesTransformer.groovy

```diff
@@ -102,39 +116,74 @@ class ConnectorAttributesTransformer {
+        attributes.put('sslProtocol', connector.getSslProtocol())
       }
       if (connector.getKeystoreFile() != null && !connector.getKeystoreFile().isEmpty()) {
-        res.put('keystoreFile', connector.getKeystoreFile())
+        attributes.put('keystoreFile', connector.getKeystoreFile())
       }
       if (connector.getKeystorePass() != null && !connector.getKeystorePass().isEmpty()) {
-        res.put('keystorePass', connector.getKeystorePass())
+        attributes.put('keystorePass', connector.getKeystorePass())
+      }
+      if (connector.getKeystoreType() != null && !connector.getKeystoreType().isEmpty()) {
+        attributes.put('keystoreType', connector.getKeystoreType())
+      }
+      if (connector.getTruststoreFile() != null && !connector.getTruststoreFile().isEmpty()) {
+        attributes.put('truststoreFile', connector.getTruststoreFile())
+      }
+      if (connector.getTruststorePass() != null && !connector.getTruststorePass().isEmpty()) {
+        attributes.put('truststorePass', connector.getTruststorePass())
+      }
+      if (connector.getTruststoreType() != null && !connector.getTruststoreType().isEmpty()) {
+        attributes.put('truststoreType', connector.getTruststoreType())
       }
       if (connector.getClientAuth() != null) {
-        res.put('clientAuth', connector.getClientAuth())
+        attributes.put('clientAuth', connector.getClientAuth())
+      }
+
+      // HTTP2
+      if (connector.getSslImplementationName() != null && !connector.getSslImplementationName().isEmpty()) {
+        attributes.put('sslImplementationName', connector.getSslImplementationName())
       }
 
       // SSL APR
       if (connector.getSslCertificateFile() != null && !connector.getSslCertificateFile().isEmpty()) {
-        res.put('SSLCertificateFile', connector.getSslCertificateFile())
+        attributes.put('SSLCertificateFile', connector.getSslCertificateFile())
+      }
+      if (connector.getSslCACertificateFile() != null && !connector.getSslCACertificateFile().isEmpty()) {
+        attributes.put('SSLCACertificateFile', connector.getSslCACertificateFile())
       }
       if (connector.getSslCertificateKeyFile() != null && !connector.getSslCertificateKeyFile().isEmpty()) {
-        res.put('SSLCertificateKeyFile', connector.getSslCertificateKeyFile())
+        attributes.put('SSLCertificateKeyFile', connector.getSslCertificateKeyFile())
       }
       if (connector.getSslPassword() != null && !connector.getSslPassword().isEmpty()) {
-        res.put('SSLPassword', connector.getSslPassword())
+        attributes.put('SSLPassword', connector.getSslPassword())
+      }
+      if (connector.getSslEnabledProtocols() != null && connector.getSslEnabledProtocols()) {
+        attributes.put('sslEnabledProtocols', connector.getSslEnabledProtocols())
+      }
+      // ---------------------
+
+
+      attributes.each { attribute ->
+        node.@"${attribute.key}" = attribute.value
```

Comment:
Could we update the `node` right away instead of `attributes`? Above.

Reply:
We can concatenate the NonSecure attributes with the secure, create a new null node(final_node) and return this.Or instead of creating new null, use the replaceNode.However its becoming more complex without any reason.

Reply:
What I ment was to update the occurs of `attributes.put('XXX', connector.getXXX())` with `node.@XXX = connector.getXXX()`. But any working solution what will not re-itarate again, will be better probably.